src/file-table-engine/src/table/format.rs

```diff
@@ -0,0 +1,215 @@
+        ))
+    }
+
+    fn build_scan_plan<T: FileOpener + Send + 'static>(
+        opener: T,
+        file_schema: Arc<Schema>,
+        files: &[String],
+        projection: Option<&Vec<usize>>,
+        limit: Option<usize>,
+    ) -> Result<PhysicalPlanRef> {
+        let stream = FileStream::new(
+            &FileScanConfig {
+                object_store_url: ObjectStoreUrl::parse("empty://").unwrap(), // won't be used
+                file_schema,
+                file_groups: vec![files
+                    .iter()
+                    .map(|filename| PartitionedFile::new(filename.to_string(), 0))
+                    .collect::<Vec<_>>()],
+                statistics: Default::default(),
+                projection: projection.cloned(),
+                limit,
+                table_partition_cols: vec![],
+                output_ordering: None,
+                infinite_source: false,
+            },
+            0, // partition: hard-code
+            opener,
+            &ExecutionPlanMetricsSet::new(),
+        )
+        .context(error::BuildStreamSnafu)?;
+        let adapter = RecordBatchStreamAdapter::try_new(Box::pin(stream))
+            .context(error::BuildStreamAdapterSnafu)?;
+        Ok(Arc::new(SimpleTableScan::new(Box::pin(adapter))))
+    }
+
+    fn new_csv_scan_plan(
+        _ctx: &CreateScanPlanContext,
+        config: &ScanPlanConfig,
+        format: &CsvFormat,
+    ) -> Result<PhysicalPlanRef> {
+        let file_schema = config.file_schema.arrow_schema().clone();
+        let opener = Format::build_csv_opener(file_schema.clone(), config, format)?;
+        Format::build_scan_plan(
+            opener,
+            file_schema,
+            config.files,
+            config.projection,
+            config.limit,
+        )
+    }
+
+    fn new_json_scan_plan(
+        _ctx: &CreateScanPlanContext,
+        config: &ScanPlanConfig,
+        format: &JsonFormat,
+    ) -> Result<PhysicalPlanRef> {
+        let file_schema = config.file_schema.arrow_schema().clone();
+        let opener = Format::build_json_opener(file_schema.clone(), config, format)?;
+        Format::build_scan_plan(
+            opener,
+            file_schema,
+            config.files,
+            config.projection,
+            config.limit,
+        )
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct ScanPlanConfig<'a> {
+    pub file_schema: SchemaRef,
+    pub files: &'a Vec<String>,
+    pub projection: Option<&'a Vec<usize>>,
+    pub filters: &'a [Expr],
+    pub limit: Option<usize>,
+    pub store: ObjectStore,
+}
+
+impl Format {
+    pub fn create_physical_plan(
+        &self,
+        ctx: &CreateScanPlanContext,
+        config: &ScanPlanConfig,
+    ) -> Result<PhysicalPlanRef> {
+        match self {
+            Format::Csv(format) => Format::new_csv_scan_plan(ctx, config, format),
+            Format::Json(format) => Format::new_json_scan_plan(ctx, config, format),
+            Format::Parquet => todo!(),
```

Comment:
Well, I wouldn't suggest use "todo!()" in our codes. Please follow our code convention about todos, and return some error like "NotSupportedSnafu" instead of panic.